CommonCompilerConfig.cmake

```diff
@@ -100,36 +100,27 @@ set(boost_required ON)
 if(NOT FORCE_CPP11)
   if(${CMAKE_CXX_COMPILER_ID} STREQUAL "GNU")
     if(CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL "9")
-         set(CMAKE_CXX_STANDARD 17)
-	 set(boost_required OFF)
+      set(CMAKE_CXX_STANDARD 17)
+      set(boost_required OFF)
     elseif(CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL "8")
-         set(CMAKE_CXX_STANDARD 17)
-	 set(extra_lib_for_filesystem "stdc++fs")
-	 set(boost_required OFF)
+      set(CMAKE_CXX_STANDARD 17)
+      set(extra_lib_for_filesystem "stdc++fs")
+      set(boost_required OFF)
     endif()	 
   elseif(${CMAKE_CXX_COMPILER_ID} MATCHES "AppleClang")  # see CMake Policy CMP0025
-    if(CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL "11") # XCode 11 & AppleClang 11 do support c++17 with <filesystem>
       ## TODO XCode11 on macOS 10.15 supports c++17 and <filesystem>,
       # https://developer.apple.com/documentation/xcode_release_notes/xcode_11_beta_5_release_notes
       # but for now CircleCI uses macOS 10.14, so we cannot disable boost yet.
       # macOS 10.15 will be release in Sept 2019, so we can switch to it soon after it.
-      set(boost_required ON)
-      set(CMAKE_CXX_STANDARD 11)
-    endif()
-  # does not support C++17 and filesystem (as of XCode 10.1)
```

Comment:
Well, OSx does not require Boost anymore with the new operating system update. Windows is already at c++17.  Linux can be c++17.

I changed it here because with the new XCode it was forcing Boost.  But I think a new PR is in order to now move up to c++17 everywhere.

Reply:
> But I think a new PR is in order to now move up to c++17 everywhere. 

definitely. But I'm happy time has come so we can finally in 2020 adhere to c++17 everywhere!